addons/website_event/models/event_event.py

```diff
@@ -116,28 +141,66 @@ def _compute_is_participating(self):
             registration;
         """
         current_visitor = self.env['website.visitor']._get_visitor_from_request()
-        base_domain = [('event_id', 'in', self.ids), ('state', 'in', ['open', 'done'])]
```

Comment:
Method docstring still mentions:

> * check only confirmed and attended registrations, a draft registration
            does not make the attendee participating;

But I don't think it does anymore? Why did we remove that part?

EDIT: Actually it seems like there is still a state condition, slightly different, and only applied when the user is not public. Not sure why.